ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -0,0 +1,201 @@
+function get_target_release() {
+    echo "RELEASE_IMAGE_INITIAL is ${RELEASE_IMAGE_INITIAL}"
+    echo "RELEASE_IMAGE_LATEST is ${RELEASE_IMAGE_LATEST}"
+    TARGET_RELEASES=("${RELEASE_IMAGE_LATEST}") &&
+    if [[ -f "${SHARED_DIR}/override-upgrade" ]]; then
+        local release_string; release_string="$(< "${SHARED_DIR}/override-upgrade")"
+        # shellcheck disable=SC2207
+        TARGET_RELEASES=($(echo "$release_string" | tr ',' ' ')) 
+        echo "Overriding upgrade target to ${TARGET_RELEASES[*]}"
+    fi
+}
+
+# Check if a release image signed
+# Parameter: 1- by-digest pull spec
+function validate_signed_image() {
+    digest="$(echo "$1" | cut -f2 -d@)"
+    algorithm="$(echo "${digest}" | cut -f1 -d:)"
+    hash="$(echo "${digest}" | cut -f2 -d:)"
+    local response; response=$(curl --silent --output /dev/null --write-out "%{http_code}" "https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release/${algorithm}=${hash}/signature-1")
+    if [[ "$response" == "200" ]]; then
+        echo "True"
+    else
+        echo "False"
+    fi
+}
+
+# Check if admin ack is required before upgrade
+# Parameters: 1-source release minor version, 2-target release minor version
+function admin_ack() {
+    set -x
+    if [[ $1 -eq $2 ]]; then
+        echo "Upgrade between z-stream version does not require admin ack"
+    else
+        local out; out="$(./oc -n openshift-config-managed get configmap admin-gates -o json | jq -r ".data")"
+        if [[ $out == *"ack-4.$1"* ]]; then
+            echo "Require admin ack"
+            local ack_data; ack_data="$(echo "$out" | cut -f2 -d\")" && echo "Admin ack patch data is: ${ack_data}"
+            if ! ./oc -n openshift-config patch configmap admin-acks --patch '{"data":{"'"${ack_data}"'": "true"}}' --type=merge; then
+                echo >&2 "Admin-acks patch command failed, exiting" && exit 1
+            fi
+            
+            echo -e "sleep 5 min wait admin-acks patch to be valid...\n"
+            local wait_time_loop_var=0
+            while(( wait_time_loop_var < 5 ));do
+                sleep 1m
+                echo -e "wait_time_passed=${wait_time_loop_var} min.\n"
+                if ! ./oc adm upgrade|grep "AdminAckRequired"; then
+                    echo -e "Admin-acks patch still in processing, waiting...\n"
+                else
+                    echo -e "Admin-acks patch finished succ\n"
+                    break
+                fi
+                ((wait_time_loop_var+=1))
+            done
+            [ "${wait_time_loop_var}" -eq 5 ] && echo >&2 "Admin-acks patch not completed in 5 minutes, exiting" && exit 1
+        fi
+    fi
+    set +x
+}
+
+# Upgrade the cluster to target release
+# Parameters: 1- by-digest pull spec, 2- by-tag pull spec
+function upgrade() {
+    set -x
+    local err
+    
+    if ! err="$(./oc adm upgrade --to-image="$1" 2>&1 >/dev/null)" && [[ $err == *"you must pass --allow-explicit-upgrade to continue"* ]]; then
```

Comment:
How about replace:
`! err="$(./oc adm upgrade --to-image="$1" 2>&1 >/dev/null)"`
to
`(! err="$(./oc adm upgrade --to-image="$1" 2>&1)")`
for easy reading.